src/main/java/com/meilisearch/sdk/Client.java

```diff
@@ -242,4 +248,58 @@ public Key createKey(Key options) throws Exception {
     public void deleteKey(String key) throws Exception {
         this.keysHandler.deleteKey(key);
     }
+
+    public String generateTenantToken(Map<String, Object> searchRules) throws MeiliSearchException {
+        return this.generateTenantToken(searchRules, new TenantTokenOptions());
+    }
+
+    /**
+     * Generate a tenant token
+     *
+     * @param searchRules A Map of string, object which contains the rules to be enforced at search
+     *     time for all or specific accessible indexes for the signing API Key.
+     * @param options A TenantTokenOptions, the following fileds are accepted: - apiKey: String
+     *     containing the API key parent of the token. If you leave it empty the client API Key will
+     *     be used. - expiresAt: A DateTime when the key will expire. Note that if an expiresAt
+     *     value is included it should be in UTC time.
+     * @return String containing the tenant token
+     * @throws MeiliSearchException if an error occurs
+     */
+    public String generateTenantToken(Map<String, Object> searchRules, TenantTokenOptions options)
+            throws MeiliSearchException {
+        // Validate all fields
+        if ((options.getApiKey() == null || options.getApiKey() == "")
+                && this.config.apiKey == "") {
```

Comment:
`this.config.apiKey` can be also `null`?

Reply:
Yes but only if you create your client like:
```
      Client client = new Client(new Config("http://localhost:7700", null));
```
Instead of:
```
      Client client = new Client(new Config("http://localhost:7700"));
```
I will ad the verification anyway